### PR TITLE
Release 0.7.0

### PR DESCRIPTION
**Pull request by chuwy**

### CODE REVIEW

**Comment by benjben** [comment]
If I'm not mistaken, data is not partitioned per hour. Should we take this opportunity to to do it, to be consistent with what's done on GCP?

File: examples/config.hocon.sample [hunk 1]
```diff
@@ -86,6 +86,8 @@ streams {
 s3 {
   region = "{{s3Region}}"
   bucket = "{{s3bucket}}"
+  # optional bucket where to store partitioned data
+  bucketJson = "{{s3bucket}}/partitioned"
```
Comment (benjben) [inline examples/config.hocon.sample]: Would be nice to precise that this needs to be specified if we want the data to be partitioned by type. Or use a boolean for that like `partitionByType`.

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala [hunk 2]
```diff
@@ -76,28 +80,52 @@ class KinesisS3Emitter(
...
-    val baseFilename = getBaseFilename(buffer.getFirstSequenceNumber, buffer.getLastSequenceNumber)
+    val records = buffer.getRecords.asScala.toList
+    val partitions = if (s3Config.bucketJson.isDefined) partitionByType(records).toList else List((RowType.Unpartitioned, records))
+
+    partitions.flatMap {
+      case (RowType.Unpartitioned, partitionRecords) =>
+        val baseFileName = getBaseFilename(buffer.getFirstSequenceNumber, buffer.getLastSequenceNumber, None)
+        emitRecords(partitionRecords,false, baseFileName)
+      case (RowType.SelfDescribing(vendor, name, model), partitionRecords) =>
+        val prefix = s"$vendor/$name/jsonschema/$model"
```
Comment (benjben) [inline src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: Why not get `jsonschema` from the SchemaKey?

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala [hunk 3]
```diff
@@ -108,3 +136,52 @@ class KinesisS3Emitter(
...
+      * Depending on partitioning setting should be either sank or sent to bad stream
+      */
+    case object Unpartitioned extends RowType
+
+    /** JSON line with self-describing payload that can be partitioned */
+    case class SelfDescribing(vendor: String, name: String, model: Int) extends RowType
+
+    /** Unrecognized line, e.g. non-string or non-SDJSON whereas partitioning is enabled */
+    case object ReadingError extends RowType
+  }
```
Comment (benjben) [inline src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: Could be in `RowType.scala`

File: src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala [hunk 4]
```diff
@@ -108,3 +136,52 @@ class KinesisS3Emitter(
...
+
+object KinesisS3Emitter {
+
+  /** Type of row which determined according to schema of self describing data */
+  sealed trait RowType extends Product with Serializable
+
+  object RowType {
+
+    /**
+      * Old-school TSV/raw line that cannot be partitioned
```
Comment (benjben) [inline src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala]: `that cannot be partitioned` => we don't always want to partition data

File: CHANGELOG [hunk 5]
```diff
@@ -1,3 +1,23 @@
+Version 0.7.0 (2019-10-15)
+--------------------------
+Update flags to resemble ones used in GCS loader config #157
```
Comment (chuwy) [inline CHANGELOG]: Missing parentheses.

**Pull request merged**